### PR TITLE
Improve usability of URL input field

### PR DESCRIPTION
**Pull request by robertknight**
This PR makes small usability improvements for the URL input field on the Via homepage following testing on iOS.
 
- Set the input type to "url" to disable unwanted autocorrect behavior and
   trigger a virtual keyboard layout specialized for entering URLs/URL-like
   strings.

   This required disabling built-in form validation via "novalidate" because the
   input accepts values (eg. domain names with no scheme) which are not accepted
   by the browser's default validation of URL inputs.

 - Autofocus the URL input field when the page loads.

 - Remove "maxsize" attribute, which is not a valid input attribute.

### CODE REVIEW

File: templates/index.html [hunk 1]
```diff
@@ -81,8 +81,16 @@
       <g fill="none" fill-rule="evenodd"><path fill="#FFF" fill-rule="nonzero" d="M3.886 3.945H21.03v16.047H3.886z"/><path d="M0 2.005C0 .898.897 0 2.005 0h19.99C23.102 0 24 .897 24 2.005v19.99A2.005 2.005 0 0 1 21.995 24H2.005A2.005 2.005 0 0 1 0 21.995V2.005zM9 24h6l-3 4-3-4zM7.008 4H4v16h3.008v-4.997C7.008 12.005 8.168 12.01 9 12c1 .007 2.019.06 2.019 2.003V20h3.008v-6.891C14.027 10 12 9.003 10 9.003c-1.99 0-2 0-2.992 1.999V4zM19 19.987c1.105 0 2-.893 2-1.994A1.997 1.997 0 0 0 19 16c-1.105 0-2 .892-2 1.993s.895 1.994 2 1.994z" fill="#3F3F3F"/></g>
     </svg>
   </a>
-  <form class="url-form" onsubmit="url = document.getElementById('search').value; if (url != '') { document.location.href = '/' + url; } return false;">
-    <input id="search" class="url-field" name="search" type="text" maxsize="30" placeholder="Paste a link to create or view annotations"/>
+  <form class="url-form"
+        onsubmit="url = document.getElementById('search').value; if (url != '') { document.location.href = '/' + url; } return false;"
+        novalidate>
+    <input id="search"
+           autofocus
```
Comment (dwolfe) [inline templates/index.html]: I had to look up autofocus accessibility, and found [this article](https://www.brucelawson.co.uk/2009/the-accessibility-of-html-5-autofocus/) (by way of a link from [this WebAIM article](https://webaim.org/blog/future-web-accessibility-html5-input-extensions/)), which for our purposes boils down to "I believe that autofocussing into a form is a usability win for most people if the purpose of the page is that form."

@lyzadanger raised the question of autofocus accessibility, so I wanted to document it here. Makes perfect sense for this page.

File: templates/index.html [hunk 2]
```diff
@@ -81,8 +81,16 @@
...
+           class="url-field"
+           name="search"
+           placeholder="Paste a link to create or view annotations"
```
Comment (dwolfe) [inline templates/index.html]: @lyzadanger also asked about labeling the search field, and we should probably move the `placeholder` text to a `label` - [this exploration](http://www.a11ymatters.com/pattern/accessible-search/) shows how screen readers interpret various search forms, and there's also the temporal availability to consider (labels are always available, placeholders only before the user enters text, tooltips only on hover, etc.).

**Comment by lyzadanger** [comment]
Given that the field didn't have a `label` before this change, I'd be happy to see improvements there as a separate PR/follow up—but I do agree with Dave! I'd like to see a `label` element for a11y.

**Comment by robertknight** [comment]
Hi Lyza and David,

Given that there is only a single input field on the form I think the placeholder makes its purpose pretty obvious for sighted users. I had a look at how eg. Google and Twitter have their search pages set up and they have either a hidden `<label>` (Twitter) or an `aria-label` attribute on the search field to make the purpose clear for others. Do you think it would make sense to do the same here?

**Comment by lyzadanger** [comment]
I'm down with the visually hidden `label` tag!

**Comment by dwolfe** [comment]
+1, yep, that'll work!

**Comment by robertknight** [comment]
I pondered this a little more and then opted for adding an `aria-label` to the field. To me that felt like a more obvious way to add a non-visual label. I'm happy to change if someone can think of a concrete issue with that versus adding a hidden `<label>`. PTAL @lyzadanger

**Comment by robertknight** [comment]
I'm taking the thumbs up as an approval here, correct me if I got that wrong!

**Pull request merged**